### ISSUE TITLE
Hangs when used with fail-fast option

### ISSUE THREAD

**Issue by stepantubanov** [cucumber-rs/cucumber#241]
I think there may be a bug with `--fail-fast` behavior. I'm assuming it should force cucumber to exit on first failure. Instead I'm seeing step failure (as expected) and afterwards it just hangs (doesn't exit, seems to be stuck in an infinite loop somewhere).

```rust
# tests/scenarios
use cucumber::{then, World as _};

#[derive(cucumber::World, Debug, Default)]
struct World;

#[then(expr = "step panics")]
async fn step_panics(_: &mut World) {
    panic!("this is a panic message");
}

#[then(expr = "nothing happens")]
async fn nothing_happens(_: &mut World) {
    // noop
}

#[tokio::main]
async fn main() {
    World::cucumber()
        .max_concurrent_scenarios(1)
        .run_and_exit("tests/check.feature")
        .await;
}
```

```gherkin
# tests/check.feature
Feature: A
  Scenario: 1
    Then step panics

  Scenario: 2
    Then nothing happens
```

Cargo.toml:
```toml
...

[dependencies]
cucumber = "0.15"
tokio = { version = "1", features = ["full"] }

[[test]]
name = "scenarios"
harness = false
```

And run like this: `cargo test --test scenarios -- --fail-fast`

**Comment by tyranron** [comment]
The problem why this hangs is [this change](https://github.com/cucumber-rs/cucumber/commit/c1d919a476ebd8f87633e49d5ca2f8b647a0fce7#diff-7cdd8dcb3b6d6d5155c767874d63aa47e2c6d97e19ca0b598bedcc6389be9a1eR2073) which makes [`is_finished` evaluate to `true` when `fail_fast` happens, so we never exit the execution loop](https://github.com/cucumber-rs/cucumber/blob/v0.15.3/src/runner/basic.rs#L923-L925).

**Comment by tyranron** [comment]
@stepantubanov released in [0.16.0](https://crates.io/crates/cucumber/0.16.0).
